### PR TITLE
Siri Shortcuts

### LINKED ISSUE #236: Add Siri Integration
Twitter user: https://twitter.com/__SteveAnderson/status/1042548421947617281

> Can we get a Siri Shortcut? “Hey Siri, add note to Simplenote” We’d be eternally grateful!

This sounds like a great enhancement although I'm not sure how easily it can be accomplished.

This was originally posted at https://github.com/Automattic/simplenote-electron/issues/868 and by request, I have moved it here.

### PR DESCRIPTION
**Pull request by jleandroperez**
### Details:
We're adding support for Siri Shortcuts for three actions (initially!):

1. Launch Simplenote
2. Add a new Note
3. Open a specific Note

@bummytime ready for a look!! thanks in advance!!
Closes #236 

### NOTE:

Exposing UserActivities by means of **becomeCurrent** fails miserably, when exposing multiple activities at once (as per... tests with the actual device). iOS either picks up the last registered activity, or none of them!

---

### Testing: Spotlight Search Sanity
1. Log into your account
2. Send Simplenote to Background
3. Search in Spotlight the contents of **any** of your notes, and tap on any of them
4. Verify Simplenote is brought to FG, and the note is onscreen

### Testing: Opening Simplenote
1. Open **Settings.app** > **Deverloper** and enable `Display recent Shortcuts`
2. Log into your account
3. Send Simplenote to BG
4. Open Spotlight
5. Verify `Open Simplenote` shows up. Tap on it, and verify Simplenote is brought to FG

### Testing: Creating a New Note
1. Open **Settings.app** > **Deverloper** and enable `Display recent Shortcuts`
2. Log into your account
3. Create a new Note, so that the new shortcut is registered
4. Open Spotlight
5. Verify `Create a new note ` shows up. Tap on it, and verify Simplenote is brought up onscreen with a blank note

### Testing: Opening a Specific Note
1. Open **Settings.app** > **Deverloper** and enable `Display recent Shortcuts`
2. Log into your account
3. Open any random note
4. Open Spotlight
5. Verify an item named `Open [Title of your note]` shows up. Tap on it, and verify the note you've opened in (3) is onscreen

### CODE REVIEW

File: Simplenote/Classes/ShortcutsHandler.swift [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        NSUserActivity.newNoteActivity(),
+        NSUserActivity.launchActivity()
+    ]
+
+    /// Registers all of the Simplenote-Y Activities.
+    ///
+    /// - Note:
+    ///     1. Calling `becomeCurrent()` sequentially causes the OS not to register anything, (OR) register
+    ///        just the last activity.
+    ///     2. Not keeping the Activities around... also causes `becomeCurrent` to fail. That's why `activities` is
```
Comment (bummytime) [inline Simplenote/Classes/ShortcutsHandler.swift]: Thanks for the explanation! 👍

File: Simplenote/Classes/ShortcutsHandler.swift [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        }
+    }
+
+    /// Removes all of the shared UserActivities, whenever the API allows.
+    ///
+    @objc
+    func unregisterSimplenoteActivities() {
+        if #available(iOS 12.0, *) {
+            NSUserActivity.deleteAllSavedUserActivities {
+                // No-Op: The SDK's API... doesn't take a nil callback. Neat!
```
Comment (bummytime) [inline Simplenote/Classes/ShortcutsHandler.swift]: 😆

File: Simplenote/SPPrivacyViewController.swift [hunk 3]
```diff
@@ -26,7 +26,8 @@ class SPPrivacyViewController: SPTableViewController {
     /// Indicates if Analytics are Enabled
     ///
     private var isAnalyticsEnabled: Bool {
-        guard let simperium = SPAppDelegate.shared()?.simperium, let preferences = simperium.preferencesObject() else {
+        let simperium = SPAppDelegate.shared().simperium
+        guard let preferences = simperium.preferencesObject() else {
```
Comment (bummytime) [inline Simplenote/SPPrivacyViewController.swift]: 🧹 👍

**Comment by jleandroperez** [comment]
Thank you for reviewing this one Matt!!

**Pull request merged**